Modules/Core/Common/include/itkExceptionObject.h

```diff
@@ -50,10 +49,15 @@ class ITKCommon_EXPORT ExceptionObject : public std::exception
 public:
   static constexpr const char * const default_exception_message = "Generic ExceptionObject";
   using Superclass = std::exception;
-  /** Various types of constructors.  Note that these functions will be
-   * called when children are instantiated.  The default constructor and
-   * the copy constructor of ExceptionObject never throw an exception. */
-  ExceptionObject() noexcept;
+
+  /** Explicitly-defaulted default-constructor. Creates an empty exception object.
+   * \note The other five "special member functions" (copy-constructor,
+   * copy-assignment operator, move-constructor, move-assignment operator,
+   * and destructor) are defaulted implicitly, following the C++ "Rule of Zero".
+   * All of these special member functions are `noexcept`.
+   */
+  ExceptionObject() noexcept = default;
+
   explicit ExceptionObject(const char * file,
```

Comment:
Since you are touching this, maybe this variant of the constructor is not necessary. `char *` is implicitly convertible into `std::string`, so the other constructor signature should be enough?

Reply:
@dzenanz For me personally, it would be fine to just remove the `const char *` `ExceptionObject` constructor overload, as you're suggesting. However, unlike the other commits of this PR, that would actually change the API!

This `itk::ExceptionObject` constructor explicitly supported "null" `char` pointers as arguments for `file`, `desc`, and `loc`:

https://github.com/InsightSoftwareConsortium/ITK/blob/v5.2rc03/Modules/Core/Common/src/itkExceptionObject.cxx#L133-L137

Would it be OK to you to drop "null" `char` pointer support for `file`, `desc`, and `loc`? And if so, how about `SetLocation` and `SetDescription`? 

    const char * nullCharPtr{ nullptr };

    itk::ExceptionObject exceptionObject(nullCharPtr, __LINE__, nullCharPtr, nullCharPtr);
    exceptionObject.SetLocation(nullCharPtr);
    exceptionObject.SetDescription(nullCharPtr);

Shall we just drop support for any of these three null-pointer "use cases"? You may also first merge this PR, and then reconsider the `const char *` issue afterwards 😸

Reply:
Let's not change the API, especially since 5.2 final is so close.